### PR TITLE
Update PDOP/HDOP/VDOP parameters in api

### LINKED ISSUE #105: [SDL 0175] Updating DOP value range for GPS notification
Proposal: [Updating DOP value range for GPS notification](https://github.com/smartdevicelink/sdl_evolution/blob/master/proposals/0175-Updating-DOP-value-range-for-GPS-notification.md)

> This proposal is to update the range and make these fields non mandatory for DOP values in GPS notification for GetVehicleData and OnVehicleData RPCs.

Review: https://github.com/smartdevicelink/sdl_evolution/issues/510

Steering Committee Decision:

> The Steering Committee voted to accept this proposal with revisions.  The revision will be to include maxValue as described in [this comment](https://github.com/smartdevicelink/sdl_evolution/issues/510#issuecomment-400474515) from the author.

>The proposal .md was updated on July 2, 2018 to include the agreed upon revisions.

### PR DESCRIPTION
**Pull request by conlain-k**
Fixes #105 by updating XML to match https://github.com/smartdevicelink/sdl_core/pull/2417

### CODE REVIEW

File: MOBILE_API.xml [hunk 1]
```diff
@@ -1688,56 +1688,104 @@
...
+        <param name="compassDirection" type="CompassDirection" mandatory="false" since="5.0">
             <description>See CompassDirection.</description>
+            <history>
+                <param name="compassDirection" type="CompassDirection" mandatory="true" since="2.0" until="5.0"/>
+            </history>
         </param>
-        <param name="pdop" type="Float" minvalue="0" maxvalue="10" defvalue="0" mandatory="true">
-            <description>PDOP.  If undefined or unavailable, then value shall be set to 0.</description>
+        <param name="pdop" type="Float" minvalue="0" maxvalue="1000" mandatory="false" since="5.0">
+            <description>PDOP.</description>
```
Comment (theresalech) [inline MOBILE_API.xml]: Per the proposal, this description should be `PDOP.  If undefined or unavailable, then value shall be set to 0.`

File: MOBILE_API.xml [hunk 2]
```diff
@@ -1688,56 +1688,104 @@
...
+            <history>
+                <param name="pdop" type="Float" minvalue="0" maxvalue="10" defvalue="0" mandatory="true" since="2.0" until="5.0"/>
+            </history>
         </param>
-        <param name="hdop" type="Float" minvalue="0" maxvalue="10" defvalue="0" mandatory="true">
-            <description>HDOP.  If value is unknown, value shall be set to 0.</description>
+        <param name="hdop" type="Float" minvalue="0" maxvalue="1000" mandatory="false" since="5.0">
+            <description>HDOP.</description>
```
Comment (theresalech) [inline MOBILE_API.xml]: Per the proposal, this description should be `HDOP.  If value is unknown, value shall be set to 0.`

File: MOBILE_API.xml [hunk 3]
```diff
@@ -1688,56 +1688,104 @@
...
+            <history>
+                <param name="hdop" type="Float" minvalue="0" maxvalue="10" defvalue="0" mandatory="true" since="2.0" until="5.0"/>
+            </history>
         </param>
-        <param name="vdop" type="Float" minvalue="0" maxvalue="10" defvalue="0" mandatory="true">
-            <description>VDOP.  If value is unknown, value shall be set to 0.</description>
+        <param name="vdop" type="Float" minvalue="0" maxvalue="1000" mandatory="false" since="5.0">
+            <description>VDOP.</description>
```
Comment (theresalech) [inline MOBILE_API.xml]: Per the proposal, this description should be `VDOP.  If value is unknown, value shall be set to 0.`

**Review by theresalech — CHANGES_REQUESTED** [review]
@JackLivio can you please review these comments and update accordingly on Conlain's behalf?  One other difference I noticed between this PR and the `Proposed solution` section of the proposal (https://github.com/smartdevicelink/sdl_evolution/blob/master/proposals/0175-Updating-DOP-value-range-for-GPS-notification.md) is that the proposal shows `dataType` and `resultCode` params within the `VehicleDataResult` struct.  It appears that those have been included in the MOBILE_API.xml since 2.0 (see: https://github.com/smartdevicelink/rpc_spec/blob/67293978702267af00f848d6d1e3f57c824cb94f/MOBILE_API.xml#L1792) so I don't think this is an issue, unless you can identify any other changes needed to that struct.  Let me know of any questions.  Thanks!

**Comment by Jack-Byrne** [comment]
@theresalech The VehicleDataResult Struct appears to be a copy and paste error. That field has existed prior to the proposal and wasn't updated

Lines in question:
```
<struct name="VehicleDataResult">
    <description>Individual published data request result</description>
    <param name="dataType" type="VehicleDataType">
      <description>Defined published data element type.</description>
    </param>
    <param name="resultCode" type="VehicleDataResultCode">
      <description>Published data result code.</description>
    </param>
  </struct>
```

**Comment by Jack-Byrne** [comment]
@theresalech updated based on comments

**Review by theresalech — APPROVED** [review]

**Comment by Jack-Byrne** [comment]
@theresalech The pdop, hdop, and vdop values do not have a default value listed in the proposal, but the description says that there should be one. I updated the PR to contain default values of 0.

**Review by joeygrover — APPROVED** [review]

**Pull request merged**